gcapi/client.py

```diff
@@ -741,3 +747,100 @@ def run_external_job(self, *, algorithm: str, inputs: Dict[str, Any]):
             job["inputs"].append(i)
 
         return (yield from self.__org_api_meta.algorithm_jobs.create(**job))
+
+    def update_archive_item(
+        self, *, archive_item_pk: str, values: Dict[str, Any]
+    ):
+        """
+        This function updates an existing archive item with the provided values
+        and returns the updated archive item.
+        You can use this function, for example, to add metadata to an archive item.
+        First, retrieve the archive items from your archive:
+        archive = next(client.archives.iterate_all(params={"slug": "..."}))
+        items = list(
+            client.archive_items.iterate_all(params={"archive": archive["id"]})
+        )
+        To then add, for example, a PDF report and a lung volume
+        value to the first archive item , do the following:
+        client.update_archive_item(
+            archive_item_pk=items[0]['id'],
+            values={
+                "report": [...],
+                "lung-volume": 1.9,
+            },
+        )
+        If you provide a value or file for an existing interface of the archive
+        item, the old value will be overwritten by the new one, hence allowing you
+        to update existing archive item values.
+        For images that are already associated with an archive item, you can
+        also change the interface type (e.g. from generic medical image to
+        generic overlay) by providing the link to the existing image together
+        with the new interface slug you would like to use:
+        client.update_archive_item(
+            archive_item_pk=items[0]['id'],
+            values={
+                "generic-overlay":
+                "https://grand-challenge.org/api/v1/cases/images/.../",
+            }
+        )
+
+        Parameters
+        ----------
+        archive_item_pk
+        values
+
+        Returns
+        -------
+        The updated archive item
+        """
+        item = yield from self.__org_api_meta.archive_items.detail(
+            pk=archive_item_pk
+        )
+        civs: Dict[str, list] = {"values": []}
+
+        for civ_title, value in values.items():
```

Comment:
Aren't these are `ci_slugs`, not `civ_titles`? We should never use titles as they're not unique.